src/main/java/net/dv8tion/jda/api/entities/IPermissionHolder.java

```diff
@@ -26,8 +26,57 @@
  *
  * @since 3.0
  */
-public interface IPermissionHolder
+public interface IPermissionHolder extends ISnowflake
 {
+    /**
+     * Converts this abstraction to the concrete {@link net.dv8tion.jda.api.entities.Role Role} type.
+     * <br>If this is not a role, this will return {@code null} instead.
+     *
+     * @return This permission holder as a concrete Role, or null
+     *
+     * @see    #isRole()
+     * @see    #asMember()
+     */
+    default Role asRole()
+    {
+        return isRole() ? (Role) this : null;
```

Comment:
This should throw, not return null.

Reply:
I actually don't know why these were added anymore, I believe the intention was to make retrieval of IDs simpler. However, since I decided to make Member extend ISnowflake that should no longer be needed